test/NotifierTest.cpp

```diff
@@ -152,7 +303,67 @@ TEST_F(NotifierTest, 1Message2ReceiversUsingSenderandReceiverThreads) {
    EXPECT_TRUE(ThreadIsShutdown(senderData));
 }
 
-TEST_F(NotifierTest, 2Messages2Receivers_RestartReceivers) {
+TEST_F(NotifierTest, 2Messages2Receivers_RestartReceivers_NoMessage) {
+   SEND_MSG_TYPE = NONE;
+   const size_t expectedFeedback = 2;
+   const size_t sendFeedback = 1;
+   TestThreadData senderData{"sender", expectedFeedback};
+   TestThreadData receiver1ThreadData{"receiver1", sendFeedback};
+   TestThreadData receiver2ThreadData{"receiver2", sendFeedback};
+   EXPECT_FALSE(FileIO::DoesFileExist(notifierQueuePath));
+   EXPECT_FALSE(FileIO::DoesFileExist(handshakeQueuePath));
+
+   // Spawn Sender and wait for it to start up
+   zthread_new(&SenderThread, reinterpret_cast<TestThreadData*>(&senderData));
+   EXPECT_TRUE(SleepUntilCondition({{senderData.hasStarted}}));
+   EXPECT_TRUE(FileIO::DoesFileExist(notifierQueuePath));
+   EXPECT_TRUE(FileIO::DoesFileExist(handshakeQueuePath));
+
+   // Spawn 2 Listeners and wait for them to start up
+   zthread_new(&ReceiverThread, reinterpret_cast<TestThreadData*>(&receiver1ThreadData));
+   zthread_new(&ReceiverThread, reinterpret_cast<TestThreadData*>(&receiver2ThreadData));
+   EXPECT_TRUE(SleepUntilCondition({{receiver1ThreadData.hasStarted}, {receiver2ThreadData.hasStarted}}));
```

Comment:
Can the tests be refactored to avoid duplication, and still be readable? The pattern seem to be the same in each test.
I.e. lifting out the common code to a function that takes the setup params